src/CallbackFilterHandler.php

```diff
@@ -81,19 +95,41 @@ public function __construct($handler, array $filters, $bubble = true)
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function pushProcessor(callable $callback): HandlerInterface
+    {
+        array_unshift($this->processors, $callback);
+        return $this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function popProcessor(): callable
+    {
+        return array_shift($this->processors);
+    }
+
     /**
      * {@inheritdoc}
      */
-    public function isHandling(array $record)
+    public function isHandling(array $record): bool
     {
-        if ($record['level'] < $this->handler->getLevel()) {
+        if ($this->handler instanceof AbstractHandler) {
```

Comment:
Would it not easier to inject the handlerlevel in the constructor? Then you dont need to care about the injected handler.

Reply:
commit f64071427af04c808d25b21a5274c685bbe2ae45 resolve it